plugins/content_distribution/providers/cross_kaltura/lib/batch/CrossKalturaDistributionEngine.php

```diff
@@ -142,12 +166,28 @@ protected function initClients(KalturaCrossKalturaDistributionProfile $distribut
 
 		// init target client
 		$targetClientConfig = new KalturaConfiguration($distributionProfile->targetAccountId);
-		$targetClientConfig->serviceUrl = $distributionProfile->targetServiceUrl;
+		$targetClientConfig->serviceUrl = $this->getTargetServiceUrl($distributionProfile->targetServiceUrl);
 		$targetClientConfig->setLogger($this);
 		$this->targetClient = new KalturaClient($targetClientConfig);
 		$ks = $this->targetClient->user->loginByLoginId($distributionProfile->targetLoginId, $distributionProfile->targetLoginPassword, $distributionProfile->targetAccountId, 86400, 'disableentitlement');
 		$this->targetClient->setKs($ks);
 	}
+	
+	private function getTargetServiceUrl($serviceUrl)
+	{
+		if(!$this->targetServiceUrlRegex)
+		{
+			return $serviceUrl;
+		}
+		
+		$matches = array();
```

Comment:
Do we need it? it is not being used as far as I can see.

Reply:
I was using it before and forgot to remove it.
Removed now.